h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -239,9 +244,11 @@ private static ParquetPreviewParseWriter readFirstRecords(ParquetParseSetup init
   }
 
   private static BlockMetaData findFirstBlock(ParquetMetadata metadata) {
+    if (metadata.getBlocks().isEmpty()) return null;
```

Comment:
We should avoid returning null here. It would be better if the caller checked the metadata before calling this function.

Reply:
Reworked this block. The layer above takes the responsibility.